src/mlir/dialect/VeronaTypes.cc

```diff
@@ -316,8 +317,11 @@ namespace mlir::verona
 
   ClassType::FieldsRef ClassType::getFields() const
   {
-    assert(getImpl()->isInitialized);
-    return getImpl()->fields;
+    // We may not have a full declaration available
```

Comment:
There was a resolved comment that explained it. Right now, we're lowering incomplete types, like S32 that haven't been declared. Until we have modules/includes, we can't declare all types and this fails to print the results.

It will be a compiler error once we have more implemented in the compiler.

Reply:
Added a `TODO` line to remind us of that later.

Reply:
I'm quite unhappy with the notion of "incomplete types". We do not want to follow the C++ model of forward declarations at all. If a type doesn't have a definition it can't be used, end of it.

Having some boilerplate empty definitions `class I32 {}` at the top of the test files is IMO a better workaround for the lack of modules/includes than introducing broken concepts into the compiler.

Reply:
This is a temporary work around and not an established behaviour. What you describe will be enforced in a more complete compiler. Trying to make the compiler perfect on day one is counter-productive.